src/visualization/gui.cc

```diff
@@ -0,0 +1,221 @@
+
+void GUI::Init() {
+  this->ecm_ = ECM::getInstance();
+
+  TEveManager::Create();
+
+  geom_ = new TGeoManager("Visualization", "Biodynamo");
+
+  // Set number of segments for approximating circles in drawing.
+  // Keep it low for better performance.
+  geom_->SetNsegments(4);
+
+  mat_empty_space_ = new TGeoMaterial("EmptySpace", 0, 0, 0);
+  mat_solid_ = new TGeoMaterial("Solid", .938, 1., 10000.);
+  med_empty_space_ = new TGeoMedium("Empty", 1, mat_empty_space_);
+  med_solid_ = new TGeoMedium("Solid", 1, mat_solid_);
+
+  // we don't know how to calculate world radius yet
+  // double worldRadius = 10000.0;
+  // top = geom_->MakeBox("World", med_empty_space_, worldRadius, worldRadius,
+  //                    worldRadius);
+
+  // Another way to make top volume for world. In this way it will be unbounded.
+  top_ = new TGeoVolumeAssembly("WORLD");
+
+  geom_->SetTopVolume(top_);
+  geom_->SetMultiThread(true);
+
+  // connect geom_ to eve
+  TGeoNode *node = geom_->GetTopNode();
+  eve_top_ = new TEveGeoTopNode(geom_, node);
+  gEve->AddGlobalElement(eve_top_);
+  gEve->AddElement(eve_top_);
+
+  // number of visualized nodes inside one volume. If you exceed this number,
+  // ROOT will draw nothing.
+  geom_->SetMaxVisNodes(max_viz_nodes_);
+
+  gEve->GetBrowser()->GetMainFrame()->SetWindowName("Biodynamo Visualization");
+
+  init_ = true;
+}
+
+void GUI::Update() {
+  if (!init_)
+    throw std::runtime_error("Call GUI::getInstance().Init() first!");
+
+  if(is_geometry_closed_)
+    throw std::runtime_error("Geometry is already closed! Don't call GUI::Update() after GUI::CloseGeometry()!");
+
+  top_->ClearNodes();
+
+  // ecm->getPhysicalSphereListCPtr() will be called only once
+  for (auto &sphere : *ecm_->getPhysicalSphereListCPtr()) {
+    auto container = new TGeoVolumeAssembly("A");
+    AddBranch(sphere, container);
+    top_->AddNode(container, top_->GetNdaughters());
+  }
+
+  gEve->FullRedraw3D();
+  gSystem->ProcessEvents();
+
+  update_ = true;
+}
+
+TGeoCombiTrans *GUI::CylinderTransformation(const PhysicalCylinder *cylinder) {
+  auto length = cylinder->getActualLength();
+  auto springAxis = cylinder->getSpringAxis();
+  auto massLocation = cylinder->getMassLocation();
+
+  auto x1 = massLocation[0];
+  auto y1 = massLocation[1];
+  auto z1 = massLocation[2];
+
+  auto dx = springAxis[0];
+  auto dy = springAxis[1];
+  auto dz = springAxis[2];
+
+  auto position = new TGeoTranslation(x1 - dx / 2, y1 - dy / 2, z1 - dz / 2);
+
+  auto phiX = 0.0, thetaY = acos(dz / length) * 180. / M_PI, psiZ = 0.0;
+
+  if ((dx < 0 && dy > 0 && dz > 0) || (dx > 0 && dy < 0 && dz < 0)) {
+    phiX = 180. - atan2(dx, dy) * 180. / M_PI;
+  } else {
+    phiX = atan2(dy, dx) * 180. / M_PI + 90.;
+  }
+
+  auto rotation = new TGeoRotation("rot", phiX, thetaY, psiZ);
+
+  return new TGeoCombiTrans(*position, *rotation);
+}
+
+EColor GUI::TranslateColor(Color color) {
+  if (color == bdm::Param::kYellow) {
```

Comment:
`switch` accepts only integers as far as I know, so type `Color` should be casted into `int`. But it has only `  bool operator==(const Color& other) const` method. 
```
  switch(color.getValue()){
    case bdm::Param::kViolet.getValue():
      return kYellow;
  }
```
Also will not work, because getValue returns non-constant unsigned int.


Going to add `operator int()` then.